src/metabase/warehouse_schema/models/table.clj

```diff
@@ -45,10 +63,21 @@
   ;; cause duplication rather than good matching if the two instances are later linked by serdes.
   #_(derive :hook/entity-id))
 
+(def ^:private transform-data-authority
+  {:out (fn [value]
+          (let [kw (some-> value keyword)]
+            (if (contains? writable-data-authority-types kw)
+              kw
+              (do (log/warnf "Unknown data_authority value from database: %s, converting to :unknown" value)
+                  :unknown))))
+   :in  #(some-> % name)})
+
 (t2/deftransforms :model/Table
   {:entity_type     mi/transform-keyword
    :visibility_type mi/transform-keyword
-   :field_order     mi/transform-keyword})
+   :field_order     mi/transform-keyword
+   ;; Warning: by using a transform to handle unexpected enum values, serialization becomes lossy
```

Comment:
Another option is to only translate unknown values to "unknown" into the API layer.

Tempted to go this route, but worried about application logic tripping over ad hoc values.

Reply:
No, I think this is the right move. It keeps things consistent across the code.